client/state/ui/layout-focus/middleware.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_FOCUS_SET,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_NEXT_FOCUS_SET,
+} from 'state/action-types';
+import { getCurrentLayoutFocus } from 'state/ui/selectors';
+import { setLayoutFocus } from 'state/ui/actions';
+import layoutFocus from 'lib/layout-focus';
+
+/**
+ * Temporary Redux middleware intended to replay dispatched Redux layout focus
+ * operations on the legacy `lib/layout-focus` module and sync legacy area
+ * changes to global state.
+ *
+ * @return {Function} Redux middleware
+ */
+export default ( { dispatch, getState } ) => {
+	layoutFocus.on( 'change', () => {
```

Comment:
Hmm, thinking again about SSR, I wonder if there are implications to registering an event listener for every Redux store instance that's initialized. Have you encountered this before @ockham ?